ingestor-default/src/main/scala/za/co/absa/hyperdrive/ingestor/implementation/reader/kafka/KafkaStreamReader.scala

```diff
@@ -129,7 +128,7 @@ object KafkaStreamReader extends StreamReaderFactory with KafkaStreamReaderAttri
     brokers.mkString(",")
   }
 
-  private def getExtraOptions(configuration: Configuration): Map[String, String] = ConfigUtils.getPropertySubset(configuration, rootFactoryOptionalConfKey)
+  private def getExtraOptions(configuration: Configuration): Map[String, String] = ConfigUtils.getPropertySubset(configuration, getExtraConfigurationPrefix.get)
```

Comment:
getExtraConfigurationPrefix.get is unsafe, but real question is why is getExtraConfigurationPrefix Optional ?

Reply:
It's optional because some components don't accept extra options. In this case, if I account for the case that `getExtraConfigurationPrefix` is `None`, I'm producing dead code.